Gradle emits unchecked warnings for gax-httpjson

```
> Task :gax-httpjson:compileJava
warning: [options] bootstrap class path not set in conjunction with -source 1.7
/usr/local/google/home/miraleung/dev/gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMethodDescriptor.java:53: warning: [unchecked] unchecked conversion
    return new AutoValue_ApiMethodDescriptor.Builder();
           ^
  required: com.google.api.gax.httpjson.ApiMethodDescriptor.Builder<RequestT,ResponseT>
  found:    com.google.api.gax.httpjson.AutoValue_ApiMethodDescriptor.Builder
  where RequestT,ResponseT are type-variables:
    RequestT extends Object declared in method <RequestT,ResponseT>newBuilder()
    ResponseT extends Object declared in method <RequestT,ResponseT>newBuilder()
/usr/local/google/home/miraleung/dev/gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java:91: warning: [unchecked] unchecked cast
        for (Object val : (List<Object>) paramValue) {
                                         ^
  required: List<Object>
  found:    Object
2 warnings

```

--- a/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java
+++ b/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMessageHttpRequestFormatter.java
@@ -75,6 +75,7 @@ public abstract class ApiMessageHttpRequestFormatter<RequestT extends ApiMessage
     return new ApiMessageHttpRequestFormatter.Builder<>();
   }
 
+  @SuppressWarnings("unchecked")
   @Override
   public Map<String, List<String>> getQueryParamNames(RequestT apiMessage) {
     Set<String> paramNames = getQueryParamNames();

--- a/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMethodDescriptor.java
+++ b/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ApiMethodDescriptor.java
@@ -50,7 +50,7 @@ public abstract class ApiMethodDescriptor<RequestT, ResponseT> {
   public abstract String getHttpMethod();
 
   public static <RequestT, ResponseT> Builder<RequestT, ResponseT> newBuilder() {
-    return new AutoValue_ApiMethodDescriptor.Builder();
+    return new AutoValue_ApiMethodDescriptor.Builder<RequestT, ResponseT>();
   }
 
   @AutoValue.Builder

